specs/transition-states.md

```diff
@@ -0,0 +1,173 @@
+# Transition states for OpenvCloud objects
+
+## Purpose
+
+For all OVC objects like **Machines**, **Cloudspaces**, **Accounts**, **Disks**, **Images**, and **Nodes** ensure that only one action is executed at a time. This is necessary to avoid race conditions caused by running actions leading to different states of an OVC object.
+
+**Static** state is a state of an OVC object, which does not change unless requested, or forced.
+**Transition** state is a state of an OVC object that occurs while transitioning between static states.
+A transition state should be introduced for all pairs of static states, that can be reached in one action.
+
+In some cases additional transition states are introduced to represent a **queued state** of an OVC object when it is being migrated. Queued state is a state that the object will switch to as soon as migration is completed.
+
+## List of transition states of OVC objects
+
+### Virtual Machine states
+
+|Initial State|Transition State| Desired State|
+|---|---|---|
+|VIRTUAL*|DEPLOYING*|RUNNING|
```

Comment:
As we discussed this will probably become VIRTUAL | DEPLOYING -> STARTING | RUNNING |
So from one transition state to the other